### ISSUE TITLE
3.1.3 dataset.data reload when doing push()

### ISSUE THREAD

**Issue by aguegu** [victorgarciaesgi/vue-chart-3#108]
## Describe the bug

https://user-images.githubusercontent.com/1559791/157365285-cc9982cc-eda4-4650-942d-c0e83286d591.mp4

As shown in the video, each data.push() action would trigger a full reload by the animation. 

3.1.3 is related to #58 , I think the cause of this problem is the cloneDeep() of data array. So each update/render is a full reload in 3.1.3

## To Reproduce

*Put a reproduction link here based on this templates*

- [Template for Vue 3](https://codesandbox.io/s/demo-vue-chart-3-ugynm)
- [Template for Vue 2](https://codesandbox.io/s/vue-chart-3-vue-2-composition-api-mw54f)
- [Template for Nuxt 2](https://codesandbox.io/s/vue-chart-3-nuxt-2-mrtej)

## Version of `vue-chart-3`
3.1.3

## Version of Vue

- [x] Vue 3
- [ ] Vue 2
- [ ] Nuxt 2
- [ ] Nuxt 3

**Comment by aguegu** [comment]
In 3.1.2, the animation only applied to the newly pushed data.

https://user-images.githubusercontent.com/1559791/157362149-d59ccbc7-7172-40e1-8794-86768d604937.mp4

**Comment by victorgarciaesgi** [comment]
I thanks for the report, looks like a rushed a little on this update. I will look at it. cloneDeep happens only if the size of the array haven't changed (Chart.js using Proxies too, I think that's was what causing the bug)

**Comment by victorgarciaesgi** [comment]
Do you reproduce it on the demo ? https://codesandbox.io/s/demo-vue-chart-3-ugynm?file=/src/App.vue

**Comment by aguegu** [comment]
To reproduce it is not hard. Just create a `line` chart with options
```js
scales: {
    x: {
      type: 'time',
      display: false,
    },
    y: {
      title: {
        display: true,
        text: 'value',
      },
    },
  },
```
and push data point to it 
```js
datasets[0].data.push({ x: Date.now(), y: Math.random() });
```
within a `setInterval()` callback. 

Hope it would help.

**Comment by GijsGoudzwaard** [comment]
I have the same issue with both a line chart and bar chart. I'm using vue-chart-3 `3.1.4` and vue `3.2.31`. For now downgrading to `3.0.9` fixes it (`npm i vue-chart-3@3.0.9`)

**Comment by victorgarciaesgi** [comment]
Yeah, it seems like the fix I added broke the animations. I need to take a time but this week it's going to be hard

**Comment by victorgarciaesgi** [comment]
Guys I'm so sorry. I just found out that when using `deep: true` using `watch` in Vue 3, the oldValue and newValue will be the same....
That's what messed with all the animations and updates

![image](https://user-images.githubusercontent.com/15092120/159270582-3fc9332f-a06e-4fe6-862a-731f275ccba0.png)

**Comment by victorgarciaesgi** [comment]
Currently finding out a way to compare new and old values

**Comment by victorgarciaesgi** [comment]
Should be fixed in `3.1.5`. Sorry again

**Comment by GijsGoudzwaard** [comment]
No worries, can confirm that this is fixed in `3.1.5`
